File: Task1Java/app/src/main/java/com/example/myfirstapp/SecondActivity.java
package com.example.myfirstapp;

import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;

public class SecondActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        // Retrieve and display the passed message
        String message = getIntent().getStringExtra("EXTRA_MESSAGE");
        TextView textView = findViewById(R.id.textViewSecond);
        textView.setText(message);

        // Set up the "Back to Main Activity" button
        Button buttonBack = findViewById(R.id.buttonBack);
        buttonBack.setOnClickListener(v -> finish()); // This will close SecondActivity and return to MainActivity
    }
}

File: Lab1/Task 1 Kotlin/app/src/main/java/com/example/myfirstapp/SecondActivity.kt
package com.example.myfirstapp

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class SecondActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        // Retrieve and display the passed message
        val message = intent.getStringExtra("EXTRA_MESSAGE")
        val textView: TextView = findViewById(R.id.textViewSecond)
        textView.text = message

        // Set up the "Back to Main Activity" button
        val buttonBack = findViewById<Button>(R.id.buttonBack)
        buttonBack.setOnClickListener {
            finish() // This will close SecondActivity and return to MainActivity
        }
    }
}
